docs/source/python/data.rst

```diff
@@ -431,4 +431,44 @@ around, so if your data is already in table form, then use
 Custom Schema and Field Metadata
 --------------------------------
 
-TODO
+The metadata used for describing the data in the schema or a field can also
```

Comment:
Maybe we can start here with a very short explanation that Arrow supports custom metadata at the schema and field level (can maybe reuse the wording from https://github.com/apache/arrow/blob/master/docs/source/format/Columnar.rst#custom-application-metadata). Something like:

> Arrow supports both schema-level and field-level custom key-value metadata allowing for systems to insert their own application defined metadata to customize behavior.

And mention this is preserved in IPC serialization or so. And then we can explain that this can be accessed at `Schema.metadata` and `Field.metadata`.